src/containers/pages/IRS/assignments/index.tsx

```diff
@@ -0,0 +1,181 @@
+import ListView from '@onaio/list-view';
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import React, { useEffect, useState } from 'react';
+import { Helmet } from 'react-helmet';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { Col, Row } from 'reactstrap';
+import { Store } from 'redux';
+import HeaderBreadcrumb from '../../../../components/page/HeaderBreadcrumb/HeaderBreadcrumb';
+import Loading from '../../../../components/page/Loading';
+import { useContextCodes } from '../../../../configs/settings';
+import {
+  ASSIGN_IRS_PLAN_URL,
+  DRAFT_IRS_PLAN_URL,
+  HOME,
+  HOME_URL,
+  IRS_PLAN_TYPE,
+  IRS_PLANS,
+  REPORT_IRS_PLAN_URL,
+} from '../../../../constants';
+import { OpenSRPService } from '../../../../services/opensrp';
+import IRSPlansReducer, {
+  reducerName as IRSPlansReducerName,
+} from '../../../../store/ducks/generic/plans';
+import {
+  extractPlanRecordResponseFromPlanPayload,
+  fetchPlanRecords,
+  getPlanRecordsArray,
+  InterventionType,
+  PlanPayload,
+  PlanRecord,
+  PlanRecordResponse,
+  PlanStatus,
+} from '../../../../store/ducks/plans';
+
+/** register the plan definitions reducer */
+reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);
+
+const OpenSrpPlanService = new OpenSRPService('plans');
+
+/** interface for PlanAssignmentsListProps props */
+interface PlanAssignmentsListProps {
+  fetchPlans: typeof fetchPlanRecords;
+  plans: PlanRecord[];
+}
+
+/** Simple component that loads plans and allows you to manage plan-jurisdiciton-organization assignments */
+const IRSAssignmentPlansList = (props: PlanAssignmentsListProps) => {
+  const { fetchPlans, plans } = props;
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const pageTitle: string = IRS_PLANS;
+
+  const breadcrumbProps = {
+    currentPage: {
+      label: pageTitle,
+      url: REPORT_IRS_PLAN_URL,
+    },
+    pages: [
+      {
+        label: HOME,
+        url: HOME_URL,
+      },
+    ],
+  };
+
+  /** async function to load the data */
+  async function loadData() {
+    try {
+      setLoading(plans.length < 1); // only set loading when there are no plans
+      OpenSrpPlanService.list()
+        .then(planResults => {
+          // filter for IRS plans
+          const irsPlans = planResults.filter((p: PlanPayload) => {
+            for (const u of p.useContext) {
+              if (u.code === useContextCodes[0]) {
+                if (u.valueCodableConcept === IRS_PLAN_TYPE) {
+                  return true;
+                } else {
+                  return false;
+                }
+              }
+            }
+            return false;
```

Comment:
we only get here if `p.useContext` is empty, is that correct?

Reply:
Yeah we only get there if for some reason the plan doesn't have the useContext for intervention type. Not sure why/how this would happen though.